### ISSUE TITLE
Feature Request: Configure O3DE to work with LGTM static analysis 

### ISSUE THREAD

**Issue by lmbr-pip** [o3de/o3de#10032]
**Is your feature request related to a problem? Please describe.**
LGTM ( a well know code static analyzer) is trying to and failing to analyze O3DE's C++ code:
https://lgtm.com/projects/g/o3de/o3de/logs/languages/lang:cpp 

Seems to be failing due to cmake mismatch:
```
[2022-06-09 23:26:34] [build-stderr] CMake Error at CMakeLists.txt:10 (cmake_minimum_required):
[2022-06-09 23:26:34] [build-stderr]   CMake 3.20 or higher is required.  You are running version 3.16.3
[2022-06-09 23:26:34] [build-stdout] -- Configuring incomplete, errors occurred!
```

**Describe the solution you'd like**
Provide custom lgtm.yml configuration file that enables LGTM to analyze O3DE's C++ Code.

**Describe alternatives you've considered**
None come to mind

**Additional context**
Unclear which SIG may own such a file so have tagged potentially interested SIGs to see if they wish to own this.

**Comment by Kadino** [comment]
It would be good to have a process defined for how static analysis findings are followed-up on or ignored as a false-positive / too-minor issue. Recommend @o3de/sig-security weigh in on this, as the stake of ownership seems to fall more in their court for driving such fixes.

**Comment by amzn-changml** [comment]
Accepting this as part of SIG-Build, but happy to work with any other SIG on this.

**Comment by lmbr-pip** [comment]
LGTM is being closed down and absorbed into GitHub actions: https://github.blog/2022-08-15-the-next-step-for-lgtm-com-github-code-scanning/

This request should really morph into a request to work with Github code scanning.

**Comment by amzn-changml** [comment]
Yes, I saw that there was a merge of their service back into Github. I've been working on doing some code scanning experiments, though it suffers the same timeout issue, even with the release build: 

https://github.com/amzn-changml/o3de/actions/runs/3275992251/jobs/5391510521
